### PR TITLE
Local test runs with data from storage

### LINKED ISSUE #3016: Load records from storage in local test runs
**Is your feature request related to a problem? Please describe.**

Make it possible to run "local test runs", that is, `ert3 run doe --local-test-run`, where `doe` is an evaluation experiment with input sources pointing to `storage`. 

**Describe the solution you'd like**
<manjana>

**Additional context**
This issue is a follow up to #2755

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Resolves #3016 

**Approach**
Remove code triggering an error in this scenario. It turns out that this works already (post recent updates)

Caveat: A record for every realization member must be uploaded prior to running a specific realization. This is a limitation of the `ert3 record load` command line API. An exception will be raised when running the experiment if the stored record ensemble has a different size.

## Pre review checklist

- [x] Added appropriate labels

Adding labels helps the maintainers when writing release notes, see sections and the
corresponding labels here: https://github.com/equinor/ert/blob/main/.github/release.yml

### CODE REVIEW

**Comment by ertomatic** [comment]
Can one of the admins verify this patch?

File: tests/ert_tests/ert3/console/integration/test_cli.py [hunk 1]
```diff
@@ -647,6 +644,24 @@ def test_cli_local_test_run_specific_realization(tmpdir):
...
+                with patch.object(
+                    sys,
+                    "argv",
+                    [
+                        "ert3",
+                        "run",
+                        "evaluation",
+                        "--local-test-run",
+                        "--realization",
+                        "200000",
```
Comment (xjules) [inline tests/ert_tests/ert3/console/integration/test_cli.py:660]: This is just nitpicking, but should `200000` be a variable than is then assert (being less than) with the number of realizations?

File: tests/ert_tests/ert3/console/integration/test_cli.py [hunk 2]
```diff
@@ -666,6 +681,75 @@ def test_cli_local_test_run_specific_realization(tmpdir):
...
+            # Modify the ensemble size in the example to avoid making 1000 http
+            # requests for the record uploading in this test:
+            ens_config = yaml.safe_load(
+                pathlib.Path("experiments/doe/ensemble.yml").read_text(encoding="utf-8")
+            )
+            ens_config["size"] = ensemble_size
+            pathlib.Path("experiments/doe/ensemble.yml").write_text(
+                yaml.dump(ens_config), encoding="utf-8"
+            )
+            args = [
```
Comment (xjules) [inline tests/ert_tests/ert3/console/integration/test_cli.py:723]: Why this needs to be loaded via loaded via `ert3 record load` and just by cfg file?

File: tests/ert_tests/ert3/console/integration/test_cli.py [hunk 3]
```diff
@@ -666,6 +681,75 @@ def test_cli_local_test_run_specific_realization(tmpdir):
...
+            args = ["ert3", "init", "--example", "polynomial"]
+            with patch.object(sys, "argv", args):
+                ert3.console._console._main()
+
+            os.chdir("polynomial")
+
+            for tainted_repo_dir in glob.glob("local-test-run*"):
+                # Because the runpath for local test runs is unpredictable,
+                # we need this in order to reliably guess where to look.
+                shutil.rmtree(tainted_repo_dir)
```
Comment (jondequinor) [inline tests/ert_tests/ert3/console/integration/test_cli.py:696]: Why do cleanup here?

File: tests/ert_tests/ert3/console/integration/test_cli.py [hunk 4]
```diff
@@ -666,6 +681,75 @@ def test_cli_local_test_run_specific_realization(tmpdir):
             assert pathlib.Path(f"local-test-run-{run_id}/output.json").exists()
 
 
+def test_cli_local_test_run_storageinput(tmpdir):
```
Comment (jondequinor) [inline tests/ert_tests/ert3/console/integration/test_cli.py:684]: Maybe this entire thing can be an example to be tested in line with the others?

File: tests/ert_tests/ert3/console/integration/test_cli.py (hunk 3 above)
Comment (berland) [reply tests/ert_tests/ert3/console/integration/test_cli.py:696]: This is mostly for the developers convenience (letting you run `pytest` without `git clean` upfront in some cases), but also serves to demonstrate the weak point in the code on unpredictable runpaths.

File: tests/ert_tests/ert3/console/integration/test_cli.py (hunk 4 above)
Comment (berland) [reply tests/ert_tests/ert3/console/integration/test_cli.py]: As in https://github.com/equinor/ert/pull/3077/commits/f62c5709f899adb91e7db94898208a9030ed1e39 ?

**Review by sondreso — APPROVED** [review]
With the previous comments addressed I think this looks good! :ship:

**Pull request merged**